wpimath/src/main/java/edu/wpi/first/math/kinematics/SwerveDriveKinematics.java

```diff
@@ -167,7 +167,7 @@ public SwerveModuleState[] toSwerveModuleStates(
       double y = moduleStatesMatrix.get(i * 2 + 1, 0);
 
       double speed = Math.hypot(x, y);
-      Rotation2d angle = new Rotation2d(x, y);
+      Rotation2d angle = speed > 1e-6 ? new Rotation2d(x, y) : m_moduleHeadings[i];
```

Comment:
```suggestion
      Rotation2d angle = Math.abs(speed) > 1e-6 ? new Rotation2d(x, y) : m_moduleHeadings[i];
```

Reply:
Wouldn't speed always be positive as it is the result of Math.hypot()?

Reply:
Yes.